### ISSUE TITLE
Images on HE are loading more slowly than usual

### ISSUE THREAD

**Issue by lostboyevsky** [jetlej/HE-6.0-Bugs#278]
I've noticed this the past few days. When I visit the site, many of the images are initially loaded only 1/3 or 1/2 of the way, or else no loaded yet at all. It takes 10-20 seconds for all images to finish loading. I'm not sure if this is due to the slowness of my computer or to some kind of change that has occurred on the site. Thought I would bring it up here. Have you seen this at all, @MSchirp or @JonBrooksX?

**Comment by JonBrooksX** [comment]
Do you know if these were featured images or images within the articles? 

Do you know how to resize images on pixelmator? We should aim to keep images well below 1MB, sub 500kb would be ideal but now always possible.

**Comment by MSchirp** [comment]
@lostboyevsky Yes, I've seen this too.

**Comment by dpatil-gmcs** [comment]
@lostboyevsky @JonBrooksX @MSchirp   I think the images needs to be optimized, I am running wp-image-smush plugin on the Production and can you please optimize the images that are uploaded on CDN server.

**Comment by lostboyevsky** [comment]
@JonBrooksX, yeah, i know how to compress image sizes in Pixelmator. i will shoot for 500kb — 1MB range from now on. 

@dpatil-gmcs, when you say, "can you please optimize the images that are uploaded on CDN server," what do you mean?

**Comment by dpatil-gmcs** [comment]
@lostboyevsky @MSchirp  By optimizing images I meant to compress the images.

**Comment by dpatil-gmcs** [comment]
@lostboyevsky @MSchirp   Any updates for us!

**Comment by lostboyevsky** [comment]
@dpatil-gmcs, I don't personally have any idea how to compress the images that you're talking about. Martijn is gone this week, so this one may have to wait until next week for us to progress further on.

**Comment by dpatil-gmcs** [comment]
@MSchirp  Can you please compress the images on CDN server? 
I have run the WP-image-Smush  plugin and It seems like it has smushed the images to some extent.

Please have a look at the image loading speed and let us know your experience.

**Comment by MSchirp** [comment]
@dpatil-gmcs Please consult with Jordan and Hayden to do this, as I don't know.

**Comment by dpatil-gmcs** [comment]
@jetlej  @haydenjames
Can you guys please look into it, The images present at your CDN server needs to be compressed. The list of images can be found on Google insights after typing 'higheixstence.com' in the text field and clicking Analyze button.

Google Insights URL: https://developers.google.com/speed/pagespeed/insights/

**Comment by dpatil-gmcs** [comment]
@MSchirp @lostboyevsky  I have compressed the images using wp Image smush, Please have a look and let me know if the image loading is better for you now.

**Comment by haydenjames** [comment]
Really didn't get the question. There's no such thing as a CDN sever which
you again mention. CDN (content delivery network) is a network of servers
strategically placed in geolocations that are not managed or accessible by
us. As I mentioned in another post you can use Cloudflare's (which is the
CDN) "Polish" service for lossless compression of images. As far as I know,
smush is an unmaintained service / script originally created by Yahoo!. As
mentioned. It's better to use Cloudflare's image compression service.

As for the the minify question. Please remind me where it was insisted that
it be disabled and why? All options should be enabled that don't conflict
with site function or user experience. Any suggestions or request to
disable any W3TC caching features would be in relation to finding bugs.
Minify is one of those options that often has to be enabled and configured
manually rather than using the auto enable setting. Cloudflare also offers
minification, don't use both, its either or as they perform the same
function.
On Apr 19, 2016 6:59 AM, "dpatil-gmcs" <notifications@github.com> wrote:

@MSchirp <https://github.com/MSchirp> @lostboyevsky
<https://github.com/lostboyevsky> I have compressed the images using wp
Image smush, Please have a look and let me know if the image loading is
better for you now.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/jetlej/HE-6.0-Bugs/issues/278#issuecomment-211933169>

**Comment by haydenjames** [comment]
Running WP plugins for compression on the server just adds unnecessary overhead when it can be done closer to end-users using Cloudflare's Polish service via their CDN. 

I would advice, not insist, that this image compression plugin be disabled and use Cloudflare. Also, config/plugin changes should be discussed between you guys before changes are made as plugins like these often result in strange issues. 

Hope this helps.

**Comment by dpatil-gmcs** [comment]
@haydenjames  By CDN we meant the images present with urls, just like '1.highexistence.com/...' etc.
We found those images in the media library of the site, so no worries about it.

And yes about minification, @MSchirp  had told me in personal during the issue #275, he asked me to disable minification as per server guys instruction.

![minification_settings](https://cloud.githubusercontent.com/assets/5171153/14676853/d843d494-072c-11e6-94fc-5abaddad8897.png)

**Comment by MSchirp** [comment]
@haydenjames FYI: That is what Jordan told me. This is why I wanted to open up lines for direct communication, since I am in the middle adding even more confusion while being confused myself.

**Comment by haydenjames** [comment]
No worries. OK so the urls 1.hiexistence.com etc use to be served via
MaxCDN. That service is no longer being used and now instead Cloudflare is
the CDN for those subdomains. (See attached screenshot)

However, the cloudflare image compression feature isn't enabled. You can
login to Cloudflare and enable the Polish feature, under the "Speed" config
options if you require lossless compression that isn't using local
CPU/resources, which also lowers bandwidth and improves end user page
speed.

With Minify I know auto enable a long time ago prevented avatars from being
edited/changed. Not sure if this is still the cause so if using the auto
then click around the site nav etc and check for any strange bugs. In which
case the .js or .CSS file would have to be manually added to the exclude
list.

Hope this is helpful.

**Comment by dpatil-gmcs** [comment]
@haydenjames  Ok then we shall enable the minification settings on the TEST site(hiexistence.com) first and observe it before enabling the minification on the LIVE site.

@lostboyevsky  @MSchirp  After smushing the images, do you guys find any difference in image loading?

**Comment by MSchirp** [comment]
@dpatil-gmcs It seems the site is still loading slower than it should. See #193 

@haydenjames How do we log in to cloud flare to follow your steps to speed up the loading of images?

**Comment by dpatil-gmcs** [comment]
@MSchirp  Yes, we are researching on #193 and trying to make the site load faster.
Also, Please let us know about the image loading speed, mentioned by @lostboyevsky  above.

**Comment by haydenjames** [comment]
Jordan has the Cloudflare credentials. I'm not sure if it's OK to post here
because there are multiple websites under that Cloudflare account.

You can enable Polish (image optimization) as per article:
https://blog.cloudflare.com/introducing-polish-automatic-image-optimizati/

It's the best option because the images are optimized and saved 'off site'
nearest to the end user via their CDN, saving CPU, bandwidth and  reducing
load times.

Thanks,

Hayden

**Comment by JonBrooksX** [comment]
Polish sounds good. What are the steps to enable it? @haydenjames

**Comment by haydenjames** [comment]
@JonBrooksX You upgrade to Cloudflare pro, then click the checkbox to enable. Cloudflare Pro is also a bit faster not just because of polish. But paid account have priority and additional CDN location propagations.
